app/pods/components/player-code-challenge/component.js

```diff
@@ -116,14 +117,40 @@ export default class CodeChallengeComponent extends Component {
   @restartableTask
   *runCodeTask (config) {
     this.set('api.headers.hackJwt', this.get('currentUser.user.hackJwt'))
-    return yield this.get("api").request("code_challenges/submit", {
+    const payload = yield this.get("api").request("code_challenges/submit", {
       method: "POST",
       data: {
         custom_input: config.input,
         source: config.source,
         language: config.lang,
       },
     });
+
+    const submissionId = payload.submissionId
+    return yield this.fetchSubmissionStatusTask(submissionId)
+  }
+  
+  /**
+   * Fetch the submission status
+   * 
+   * @param {*} submissionId 
+   * @returns {Promise} API result Promise
+   */
+
+  async fetchSubmissionStatusTask (submissionId) {
+    let maxTries = 30
+    const gap = 2000
+    let submissionStatus
+    while (maxTries--) {
```

Comment:
why not use `taskPoller` service ?

Reply:
@jatinkatyal13 and I tried to add it but the taskPoller service didn't work! Our option is either to fix it or not use it.

Reply:
@ananay It is already used elsewhere in this project. I say fix it and not rewrite logic for every polling

Reply:
Done